### PR TITLE
Add a material design checkbox

### PR DESCRIPTION
**Pull request by jneidlinger**
Google doc:

http://www.google.com/design/spec/components/switches.html#switches-checkbox

Screenshot:

image

### CODE REVIEW

**Comment by jneidlinger** [comment]
Fixes issue #47

File: modules/Material/CheckBox.qml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import QtQuick 2.0
+import QtQuick.Controls 1.2 as Controls
+import QtQuick.Controls.Styles 1.2
+import Material 0.1
+
+Controls.CheckBox {
+    id: checkBox
+
+    /*!
+       The switch color. By default this is the app's accent color
```
Comment (iBeliever) [inline modules/Material/CheckBox.qml]: Can you update this to "The checkbox color"?

File: modules/Material/CheckBox.qml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+     */
+    property color color: darkBackground ? Theme.dark.accentColor : Theme.light.accentColor
+
+    /*!
+       Set to \c true if the switch is on a dark background
```
Comment (iBeliever) [inline modules/Material/CheckBox.qml]: Checkbox here as well

File: modules/Material/qmldir [hunk 3]
```diff
@@ -8,6 +8,7 @@ AwesomeIcon 0.1 AwesomeIcon.qml
 Button 0.1 Button.qml
 Card 0.1 Card.qml
 CardPage 0.1 CardPage.qml
+CheckBox 0.1 CheckBox.qml
```
Comment (iBeliever) [inline modules/Material/qmldir]: Can you rename CheckBox to Checkbox (lowercase B) to match the Google spec?

**Comment by jneidlinger** [comment]
@iBeliever fixed :)

**Comment by iBeliever** [comment]
Great work on the new component. Could you also add support for an optional label, like the radio button component has?

**Comment by jneidlinger** [comment]
@iBeliever Already has optional label support, I added it to the demo now.

image

**Comment by jneidlinger** [comment]
@iBeliever I had to fix up some of the file-name casings, should be good now

**Pull request closed without merging**